pkg/jaeger/store/translation.go

```diff
@@ -79,3 +82,156 @@ func jSpanKindToInternal(spanKind string) string {
 	}
 	return "unspecified"
 }
+
+func ProtoToTraces(span *model.Span) (ptrace.Traces, error) {
+	encodeBinaryTags(span)
+	batches := []*model.Batch{
+		{
+			Spans: []*model.Span{span},
+		},
+	}
+	traces, err := jaegertranslator.ProtoToTraces(batches)
+	if err != nil {
+		return ptrace.NewTraces(), err
+	}
+
+	// TODO: There's an open PR against the Jaeger translator that adds support
+	// for keeping the RefType. Once the PR is merged we can remove the following
+	// if condition and the addRefTypeAttributeToLinks function.
+	//
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14463
+	if len(span.References) > 1 {
+		links := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0).Links()
+		addRefTypeAttributeToLinks(span, links)
+	}
+
+	return traces, nil
+}
+
+// TODO: There's an open PR against the Jaeger translator that adds support
+// for keeping the RefType. Once the PR is merged we can delete this function.
+//
+// https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14463
+//
+// addRefTypeAttributeToLinks adds the RefType of the Jaeger Span references as
+// an attribute to their corresponding OTEL links. The `links` argument must
+// be the OTEL representation of the given `span.References`.
+//
+// The added attributes follow the OpenTracing to OTEL semantic convention
+// https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/compatibility/#opentracing
+func addRefTypeAttributeToLinks(span *model.Span, links ptrace.SpanLinkSlice) {
+
+	// The reference to the parent span is stored directly as an attribute
+	// of the Span and not as a Link.
+	parentsSpanID := span.ParentSpanID()
+	otherParentsSpanIDs := make(map[model.SpanID]struct{})
```

Comment:
nit: `otherParentsSpanID`s can be a stack variable
```suggestion
	otherParentsSpanIDs := map[model.SpanID]struct{}
```

Reply:
I thought that using make or the literal form where basically the same, except that the later lets you specify some initial values while the former lets you specify an initial size.